borg/key.py

```diff
@@ -189,7 +190,11 @@ def env_passphrase(cls, default=None):
 
     @classmethod
     def getpass(cls, prompt):
-        return cls(getpass.getpass(prompt))
+        askpass = os.environ.get('BORG_ASKPASS', None)
+        if askpass is None:
+            return cls(getpass.getpass(prompt))
+        else:
+            return cls(check_output([askpass, prompt]).decode('UTF-8').strip())
```

Comment:
Since BORG_PASSPHRASE comes from os.environ the only consistent choice would be sys.getfilesystemencoding() + surrogateescape imho